### DISCUSSION TITLE
Openssl doesnt recognice oqs

### DISCUSSION THREAD

**Discussion by Commander99** [openssl/openssl#25732] [Q&A] [answered]
└─# make -j$(nproc)

/usr/bin/perl util/mkinstallvars.pl PREFIX=. BINDIR=apps APPLINKDIR=ms LIBDIR= INCLUDEDIR=include "INCLUDEDIR=./include" ENGINESDIR=engines MODULESDIR=providers "VERSION=3.5.0-dev" "LDLIBS=-ldl -pthread " > builddata.pm
/usr/bin/perl "-I." "-Iutil/perl" "-Mconfigdata" "-MOpenSSL::paramnames" "util/dofile.pl" "-oMakefile" crypto/params_idx.c.in > crypto/params_idx.c
/usr/bin/perl util/mkinstallvars.pl "PREFIX=/usr/local" BINDIR=bin "LIBDIR=lib64" "libdir=/usr/local/lib64" INCLUDEDIR=include APPLINKDIR=include/openssl "ENGINESDIR=/usr/local/lib64/engines-3" "MODULESDIR=/usr/local/lib64/ossl-modules" "PKGCONFIGDIR=/usr/local/lib64/pkgconfig" "CMAKECONFIGDIR=/usr/local/lib64/cmake/OpenSSL" "LDLIBS=-ldl -pthread " "VERSION=3.5.0-dev" > installdata.pm
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/crypto/bn_conf.h.in > include/crypto/bn_conf.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/crypto/dso_conf.h.in > include/crypto/dso_conf.h
/usr/bin/perl "-I." "-Iutil/perl" "-Mconfigdata" "-MOpenSSL::paramnames" "util/dofile.pl" "-oMakefile" include/internal/param_names.h.in > include/internal/param_names.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/asn1.h.in > include/openssl/asn1.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/asn1t.h.in > include/openssl/asn1t.h
DEBUG: all keys: APPLINKDIR, BINDIR, CMAKECONFIGDIR, ENGINESDIR, INCLUDEDIR, LDLIBS, LIBDIR, MODULESDIR, PKGCONFIGDIR, PREFIX, VERSION, libdir
DEBUG: PREFIX = /usr/local => PREFIX = /usr/local
DEBUG: libdir = /usr/local/lib64 => libdir = /usr/local/lib64
DEBUG: BINDIR = bin => BINDIR = /usr/local/bin, BINDIR_REL_PREFIX = bin
DEBUG: LIBDIR = lib64 => LIBDIR = /usr/local/lib64, LIBDIR_REL_PREFIX = lib64
DEBUG: INCLUDEDIR = include => INCLUDEDIR = /usr/local/include, INCLUDEDIR_REL_PREFIX = include
DEBUG: APPLINKDIR = include/openssl => APPLINKDIR = /usr/local/include/openssl, APPLINKDIR_REL_PREFIX = include/openssl
DEBUG: ENGINESDIR = /usr/local/lib64/engines-3 => ENGINESDIR = /usr/local/lib64/engines-3, ENGINESDIR_REL_LIBDIR = engines-3
DEBUG: MODULESDIR = /usr/local/lib64/ossl-modules => MODULESDIR = /usr/local/lib64/ossl-modules, MODULESDIR_REL_LIBDIR = ossl-modules
DEBUG: PKGCONFIGDIR = /usr/local/lib64/pkgconfig => PKGCONFIGDIR = /usr/local/lib64/pkgconfig, PKGCONFIGDIR_REL_LIBDIR = pkgconfig
DEBUG: CMAKECONFIGDIR = /usr/local/lib64/cmake/OpenSSL => DEBUG: all keys: CMAKECONFIGDIR = /usr/local/lib64/cmake/OpenSSL, CMAKECONFIGDIR_REL_LIBDIR = cmake/OpenSSLAPPLINKDIR, BINDIR, CMAKECONFIGDIR, ENGINESDIR, INCLUDEDIR, LDLIBS, LIBDIR, MODULESDIR, PKGCONFIGDIR, PREFIX, VERSION, libdir

No value given for CMAKECONFIGDIR
No value given for PKGCONFIGDIR
No value given for libdir
DEBUG: PREFIX = . => PREFIX = /root/openssl
DEBUG: libdir = . => libdir = /root/openssl
DEBUG: BINDIR = apps => BINDIR = /root/openssl/apps, BINDIR_REL_PREFIX = apps
DEBUG: LIBDIR =  => LIBDIR = /root/openssl, LIBDIR_REL_PREFIX = 
DEBUG: INCLUDEDIR = [ include, ./include ] => INCLUDEDIR = [ /root/openssl/include, /root/openssl/include ], INCLUDEDIR_REL_PREFIX = [ include, ./include ]
DEBUG: APPLINKDIR = ms => APPLINKDIR = /root/openssl/ms, APPLINKDIR_REL_PREFIX = ms
DEBUG: ENGINESDIR = engines => ENGINESDIR = /root/openssl/engines, ENGINESDIR_REL_LIBDIR = engines
DEBUG: MODULESDIR = providers => MODULESDIR = /root/openssl/providers, MODULESDIR_REL_LIBDIR = providers
DEBUG: PKGCONFIGDIR = . => PKGCONFIGDIR = /root/openssl, PKGCONFIGDIR_REL_LIBDIR = .
DEBUG: CMAKECONFIGDIR = . => CMAKECONFIGDIR = /root/openssl, CMAKECONFIGDIR_REL_LIBDIR = .
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/bio.h.in > include/openssl/bio.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/cmp.h.in > include/openssl/cmp.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/cms.h.in > include/openssl/cms.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/comp.h.in > include/openssl/comp.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/conf.h.in > include/openssl/conf.h
/usr/bin/perl "-I." "-Iutil/perl" "-Mconfigdata" "-MOpenSSL::paramnames" "util/dofile.pl" "-oMakefile" include/openssl/core_names.h.in > include/openssl/core_names.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/crmf.h.in > include/openssl/crmf.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/crypto.h.in > include/openssl/crypto.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/ct.h.in > include/openssl/ct.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/err.h.in > include/openssl/err.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/ess.h.in > include/openssl/ess.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/fipskey.h.in > include/openssl/fipskey.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/lhash.h.in > include/openssl/lhash.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/ocsp.h.in > include/openssl/ocsp.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/opensslv.h.in > include/openssl/opensslv.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/pkcs12.h.in > include/openssl/pkcs12.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/pkcs7.h.in > include/openssl/pkcs7.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/safestack.h.in > include/openssl/safestack.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/srp.h.in > include/openssl/srp.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/ssl.h.in > include/openssl/ssl.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/ui.h.in > include/openssl/ui.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/x509.h.in > include/openssl/x509.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/x509_acert.h.in > include/openssl/x509_acert.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/x509_vfy.h.in > include/openssl/x509_vfy.h
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" include/openssl/x509v3.h.in > include/openssl/x509v3.h
/usr/bin/perl "-I." "-Mconfigdata" "-Mbuilddata" "util/dofile.pl" "-oMakefile" exporters/pkg-config/libcrypto.pc.in > libcrypto.pc
/usr/bin/perl "-I." "-Mconfigdata" "-Mbuilddata" "util/dofile.pl" "-oMakefile" exporters/pkg-config/libssl.pc.in > libssl.pc
/usr/bin/perl "-I." "-Mconfigdata" "util/dofile.pl" "-oMakefile" test/provider_internal_test.cnf.in > test/provider_internal_test.cnf
/usr/bin/perl "-I." "-Mconfigdata" "-Mbuilddata" "util/dofile.pl" "-oMakefile" exporters/cmake/OpenSSLConfig.cmake.in > OpenSSLConfig.cmake
/usr/bin/perl "-I." "-Mconfigdata" "-Minstalldata" "util/dofile.pl" "-oMakefile" exporters/cmake/OpenSSLConfig.cmake.in > exporters/OpenSSLConfig.cmake
/usr/bin/perl "-I." "-Mconfigdata" "-Minstalldata" "util/dofile.pl" "-oMakefile" exporters/pkg-config/libcrypto.pc.in > exporters/libcrypto.pc
/usr/bin/perl "-I." "-Mconfigdata" "-Minstalldata" "util/dofile.pl" "-oMakefile" exporters/pkg-config/libssl.pc.in > exporters/libssl.pc
/usr/bin/perl "-I." "-Mconfigdata" "-Mconfigdata" "-Minstalldata" "util/dofile.pl" "-oMakefile" exporters/cmake/OpenSSLConfigVersion.cmake.in > exporters/OpenSSLConfigVersion.cmake
/usr/bin/perl "-I." "-Mconfigdata" "-Mbuilddata" "-Mbuilddata" "-Mbuilddata" "util/dofile.pl" "-oMakefile" exporters/pkg-config/openssl.pc.in > openssl.pc
/usr/bin/perl "-I." "-Mconfigdata" "-Mconfigdata" "-Mconfigdata" "-Minstalldata" "util/dofile.pl" "-oMakefile" exporters/pkg-config/openssl.pc.in > exporters/openssl.pc
/usr/bin/perl "-I." "-Mconfigdata" "-Mconfigdata" "-Mbuilddata" "util/dofile.pl" "-oMakefile" exporters/cmake/OpenSSLConfigVersion.cmake.in > OpenSSLConfigVersion.cmake
"make" depend && "make" _build_sw
make[1]: Entering directory '/root/openssl'
make[1]: Leaving directory '/root/openssl'
make[1]: Entering directory '/root/openssl'
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_libctx.d.tmp -c -o apps/lib/libapps-lib-app_libctx.o apps/lib/app_libctx.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_params.d.tmp -c -o apps/lib/libapps-lib-app_params.o apps/lib/app_params.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_provider.d.tmp -c -o apps/lib/libapps-lib-app_provider.o apps/lib/app_provider.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_rand.d.tmp -c -o apps/lib/libapps-lib-app_rand.o apps/lib/app_rand.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_x509.d.tmp -c -o apps/lib/libapps-lib-app_x509.o apps/lib/app_x509.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-apps.d.tmp -c -o apps/lib/libapps-lib-apps.o apps/lib/apps.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-apps_opt_printf.d.tmp -c -o apps/lib/libapps-lib-apps_opt_printf.o apps/lib/apps_opt_printf.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-apps_ui.d.tmp -c -o apps/lib/libapps-lib-apps_ui.o apps/lib/apps_ui.c
gcc: error: unrecognized command-line option '--with-oqs'
make[1]: *** [Makefile:4663: apps/lib/libapps-lib-apps_opt_printf.o] Error 1
make[1]: *** Waiting for unfinished jobs....
gcc: error: unrecognized command-line option '--with-oqs'
gcc: error: unrecognized command-line option '--with-oqs'
make[1]: *** [Makefile:4615: apps/lib/libapps-lib-app_libctx.o] Error 1
gcc: error: unrecognized command-line option '--with-oqs'
gcc: error: unrecognized command-line option '--with-oqs'
make[1]: *** [Makefile:4623: apps/lib/libapps-lib-app_params.o] Error 1
gcc: error: unrecognized command-line option '--with-oqs'
gcc: error: unrecognized command-line option '--with-oqs'
make[1]: *** [Makefile:4639: apps/lib/libapps-lib-app_rand.o] Error 1
make[1]: *** [Makefile:4631: apps/lib/libapps-lib-app_provider.o] Error 1
make[1]: *** [Makefile:4647: apps/lib/libapps-lib-app_x509.o] Error 1
make[1]: *** [Makefile:4655: apps/lib/libapps-lib-apps.o] Error 1
gcc: error: unrecognized command-line option '--with-oqs'
make[1]: *** [Makefile:4671: apps/lib/libapps-lib-apps_ui.o] Error 1
make[1]: Leaving directory '/root/openssl'
make: *** [Makefile:2957: build_sw] Error 2
                                                                                                                                                                     
┌──(root㉿kali)-[~/openssl]
└─# make -j$(nproc)

"make" depend && "make" _build_sw
make[1]: Entering directory '/root/openssl'
make[1]: Leaving directory '/root/openssl'
make[1]: Entering directory '/root/openssl'
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_libctx.d.tmp -c -o apps/lib/libapps-lib-app_libctx.o apps/lib/app_libctx.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_params.d.tmp -c -o apps/lib/libapps-lib-app_params.o apps/lib/app_params.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_provider.d.tmp -c -o apps/lib/libapps-lib-app_provider.o apps/lib/app_provider.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_rand.d.tmp -c -o apps/lib/libapps-lib-app_rand.o apps/lib/app_rand.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-app_x509.d.tmp -c -o apps/lib/libapps-lib-app_x509.o apps/lib/app_x509.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-apps.d.tmp -c -o apps/lib/libapps-lib-apps.o apps/lib/apps.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-apps_opt_printf.d.tmp -c -o apps/lib/libapps-lib-apps_opt_printf.o apps/lib/apps_opt_printf.c
gcc  -I. -Iinclude -Iapps/include  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O3 --with-oqs -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib64/engines-3\"" -DMODULESDIR="\"/usr/local/lib64/ossl-modules\"" -DOPENSSL_BUILDING_OPENSSL -DNDEBUG  -MMD -MF apps/lib/libapps-lib-apps_ui.d.tmp -c -o apps/lib/libapps-lib-apps_ui.o apps/lib/apps_ui.c
gcc: error: unrecognized command-line option '--with-oqs'
gcc: error: unrecognized command-line option '--with-oqs'
gcc: error: unrecognized command-line option '--with-oqs'
gcc: error: unrecognized command-line option '--with-oqs'
make[1]: *** [Makefile:4615: apps/lib/libapps-lib-app_libctx.o] Error 1
gcc: error: unrecognized command-line option '--with-oqs'
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:4623: apps/lib/libapps-lib-app_params.o] Error 1
make[1]: *** [Makefile:4639: apps/lib/libapps-lib-app_rand.o] Error 1
gcc: error: unrecognized command-line option '--with-oqs'
make[1]: *** [Makefile:4631: apps/lib/libapps-lib-app_provider.o] Error 1
make[1]: *** [Makefile:4647: apps/lib/libapps-lib-app_x509.o] Error 1
make[1]: *** [Makefile:4655: apps/lib/libapps-lib-apps.o] Error 1
gcc: error: unrecognized command-line option '--with-oqs'
gcc: error: unrecognized command-line option '--with-oqs'
make[1]: *** [Makefile:4663: apps/lib/libapps-lib-apps_opt_printf.o] Error 1
make[1]: *** [Makefile:4671: apps/lib/libapps-lib-apps_ui.o] Error 1
make[1]: Leaving directory '/root/openssl'
make: *** [Makefile:2957: build_sw] Error 2

while building openssl with liboqs i get this error

**Comment by mattcaswell** [comment, marked as answer]
This is not the correct way to use liboqs with OpenSSL.

Just build and install OpenSSL normally (or use the system installed OpenSSL if you have one).

You then need to use the oqs-provider to extend OpenSSL with liboqs support. Look at the various .md files and instructions for oqs-provider here:

https://github.com/open-quantum-safe/oqs-provider
